### PR TITLE
emilua: init at 0.4.3

### PR DESCRIPTION
**Pull request by manipuladordedados**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/emilua/default.nix [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+in
+stdenv.mkDerivation rec {
+  pname = "emilua";
+  version = "0.4.3";
+
+  src = fetchFromGitLab {
+      owner = "emilua";
+      repo = "emilua";
+      rev = "v${version}";
+      name = pname;
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:49]:
```suggestion
```

No need to set this.

File: pkgs/development/interpreters/emilua/default.nix [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    liburing
+    openssl
+    meson
+    ninja
+  ];
+
+  # Meson is no longer able to pick up Boost automatically.
+  # https://github.com/NixOS/nixpkgs/issues/86131
+  BOOST_INCLUDEDIR = "${lib.getDev boost182}/include";
+  BOOST_LIBRARYDIR = "${lib.getLib boost182}/lib";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:80]:
```suggestion
  env = {
    BOOST_INCLUDEDIR = "${lib.getDev boost182}/include";
    BOOST_LIBRARYDIR = "${lib.getLib boost182}/lib";
  };
```

File: pkgs/development/interpreters/emilua/default.nix [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+  mesonFlags = [
+    "-Dversion_suffix=-nix1"
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:83]: Why does this specific flag do?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/emilua/default.nix (hunk 3 above)
Comment (manipuladordedados) [reply pkgs/development/interpreters/emilua/default.nix:83]: Specification to emilua on which system it was built/is running.

Comment (AndersonTorres) [reply pkgs/development/interpreters/emilua/default.nix:84]: Is this needed by any reason?

Comment (manipuladordedados) [reply pkgs/development/interpreters/emilua/default.nix:84]: Yep, for debug purposes.

File: pkgs/development/interpreters/emilua/default.nix [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  # Meson is no longer able to pick up Boost automatically.
+  # https://github.com/NixOS/nixpkgs/issues/86131
+  env = {
+    BOOST_INCLUDEDIR = "${lib.getDev boost182}/include";
+    BOOST_LIBRARYDIR = "${lib.getLib boost182}/lib";
+  };
+
+  mesonFlags = [
+    "-Dversion_suffix=-nix1"
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:84]: That being said

```suggestion
    "-Dversion_suffix=-nixpkgs1"
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Squash the commits according to the contributing guide.

**Comment by AndersonTorres** [comment]
Squash the three last commits into only one, in order to comply with the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

**Comment by manipuladordedados** [comment]
> Squash the commits according to the contributing guide.

Done.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/development/interpreters/emilua/default.nix [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    "-Denable_io_uring=true"
+    "-Denable_linux_namespaces=true"
+    "-Denable_tests=true"
+    "-Denable_manpages=true"
+  ];
+
+  postPatch = ''
+    pushd subprojects
+    cp -r ${emilua-http-wrap} ${emilua-http-wrap.name}
+    cp -r ${trial-protocol-wrap} ${trial-protocol-wrap.name}
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:96]:
```suggestion
    cp -r ${emilua-http-wrap} emilua-http
    cp -r ${trial-protocol-wrap} trial-protocol
```

File: pkgs/development/interpreters/emilua/default.nix [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    cp "packagefiles/emilua-http/meson.build" "emilua-http/"
+    cp "packagefiles/trial.protocol/meson.build" "trial-protocol/"
+    popd
+
+    substituteInPlace src/emilua_gperf.awk  --replace '#!/usr/bin/env -S gawk --file' '#!${gawk}/bin/gawk -f'
+  '';
+
+  meta = with lib; {
+    description = "Lua execution engine";
+    homepage = "https://gitlab.com/emilua/emilua";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:107]:
```suggestion
    homepage = "https://emilua.org/";
```

File: pkgs/development/interpreters/emilua/default.nix [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  version = "0.4.3";
+
+  src = fetchFromGitLab {
+      owner = "emilua";
+      repo = "emilua";
+      rev = "v${version}";
+      hash = "sha256-vZITPQ1qUHhw24c0HKdR6VenviOc6JizQQ8w7K94irc=";
+  };
+
+  buildInputs = [
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:52]: What was your reasoning to fill the `buildInputs` and  `nativeBuildInputs` lists? They look mixed and random.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/emilua/default.nix [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    luajit_openresty
+    re2c
+    gperf
+    gawk
+    xxd
+    pkg-config
+    asciidoctor
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:59]: Those should be in `nativeBuildInputs`

    gawk
    xxd
    pkg-config
    asciidoctor

File: pkgs/development/interpreters/emilua/default.nix [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  nativeBuildInputs = [
+    boost182
+    fmt
+    luajit_openresty
+    ncurses
+    serd
+    sord
+    libcap
+    liburing
+    openssl
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:71]: Those should be in buildInputs:

    boost182
    fmt
    luajit_openresty
    ncurses
    serd
    sord
    libcap
    liburing
    openssl

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/emilua/default.nix (hunk 7 above)
Comment (manipuladordedados) [reply pkgs/development/interpreters/emilua/default.nix:52]: I got confused based on the template of other packages.

File: pkgs/development/interpreters/emilua/default.nix [hunk 10]
```diff
@@ -53,22 +53,21 @@ stdenv.mkDerivation rec {
     luajit_openresty
     re2c
     gperf
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:55]: re2c and gperf go to nativeBuildInputs too- according to other packages on Nixpkgs

File: pkgs/development/interpreters/emilua/default.nix (hunk 7 above)
Comment (AndersonTorres) [reply pkgs/development/interpreters/emilua/default.nix:52]: This is a bit harder to understand and it's linked to cross compilation.
Summarizing, `nativeBuildInputs` are used when the software is built, while `buildInputs` are used when the software is running.

https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-dependencies

You can look at the Nixpkgs source code to pick some examples.

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**